### PR TITLE
TELCODOCS-386: Send SRO metrics as telemetry data

### PR DESCRIPTION
**Pull request by cherraz**
Adds a module with the new metrics. Split 2 procedures to provide verification steps, and added a tip to explain reader metrics could be useful when installing.

Documents: https://issues.redhat.com/browse/TELCODOCS-386

Preview links:

The new module --> https://deploy-preview-41158--osdocs.netlify.app/openshift-enterprise/latest/hardware_enablement/psap-special-resource-operator.html#special-resource-operator-metrics_special-resource-operator

Modules slightly edited --> 
https://deploy-preview-41158--osdocs.netlify.app/openshift-enterprise/latest/hardware_enablement/psap-special-resource-operator.html#deploy-simple-kmod-using-configmap-chart

https://deploy-preview-41158--osdocs.netlify.app/openshift-enterprise/latest/hardware_enablement/psap-special-resource-operator.html#deploy-simple-kmod-using-local-chart_special-resource-operator

https://deploy-preview-41327--osdocs.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp-deploying-disconnected#ztp-checking-the-managed-cluster-status_ztp-deploying-disconnected

Versions:  4.10

### CODE REVIEW

File: modules/psap-special-resource-operator-using-configmaps.adoc [hunk 1]
```diff
@@ -310,7 +318,7 @@ simple_procfs_kmod     16384  0
 simple_kmod            16384  0
 ----
 
-[NOTE]
-====
-If you want to remove the simple-kmod kernel module from the node, delete the simple-kmod `SpecialResource` API object using the `oc delete` command. The kernel module is unloaded when the driver container pod is deleted.
-====
+[TIP]
+===
```
Comment (cherraz) [inline modules/psap-special-resource-operator-using-configmaps.adoc:322]: tips seem to be missing one =

**Comment by cherraz** [comment]
Hi @yevgeny-shnaidman , Please have a look at the PR and let me know if anything needs changing or more explanation. Thank you!

File: modules/psap-special-resource-operator-metrics.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+|Metric Name |Description
+
+|`sro_used_nodes`
+|Returns the nodes that are running pods created by a SRO custom resource (CR). This metric is available for `DaemonSet` and `Deployment` objects only. 
+
+|`sro_kind_completed_info`
+|Represents whether a `kind` of SRO CR object has been successfully uploaded in the cluster (value `1`) or not (value `0`). Examples of objects are `DaemonSet`, `Deployment` or `BuildConfig`.
+
+|`sro_states_completed_info`
+|Represents whether the SRO has finished processing a CR successfully (value `1`) or not (value `0`).
```
Comment (yevgeny-shnaidman) [inline modules/psap-special-resource-operator-metrics.adoc:21]: value 0 means that CR has not been processed yet.  Operator will always try to process a CR until successful completion. If there is an error, then a CR will be rescheduled

File: modules/psap-special-resource-operator-metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+The Special Resource Operator (SRO) exposes the following Prometheus metrics through the `metrics` service:
+
+|===
+|Metric Name |Description
+
+|`sro_used_nodes`
+|Returns the nodes that are running pods created by a SRO custom resource (CR). This metric is available for `DaemonSet` and `Deployment` objects only. 
+
+|`sro_kind_completed_info`
+|Represents whether a `kind` of SRO CR object has been successfully uploaded in the cluster (value `1`) or not (value `0`). Examples of objects are `DaemonSet`, `Deployment` or `BuildConfig`.
```
Comment (yevgeny-shnaidman) [inline modules/psap-special-resource-operator-metrics.adoc:18]: maybe:
"whether a `kind` of an object defined by the Helm charts of SRO CR?
SRO CR itself is an object. It contains pointer to helm chart/s. Inside Helm charts there is a definition of one of more objects to be uploaded. "Kind" refers to those objects

File: modules/psap-special-resource-operator-metrics.adoc [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/psap-special-resource-operator.adoc
+
+:_content-type: REFERENCE
+[id="special-resource-operator-metrics_{context}"]
+= Prometeus Special Resource Operator metrics
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-metrics.adoc:7]:
```suggestion
= Prometheus Special Resource Operator metrics
```

File: modules/psap-special-resource-operator-metrics.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/psap-special-resource-operator.adoc
+
+:_content-type: REFERENCE
+[id="special-resource-operator-metrics_{context}"]
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-metrics.adoc:6]: Ideally, your ID statement should match your filename.

File: modules/psap-special-resource-operator-metrics.adoc [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+|Represents whether a `kind` of an object defined by the Helm Charts in a SRO CR has been successfully uploaded in the cluster (value `1`) or not (value `0`). Examples of objects are `DaemonSet`, `Deployment` or `BuildConfig`.
+
+|`sro_states_completed_info`
+|Represents whether the SRO has finished processing a CR successfully (value `1`) or the SRO has not processed the CR yet (value `0`).
+
+|`sro_managed_resources_total`
+|Returns the number of SRO CRs in the cluster, regardless of their state.
+|===
+
+
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-metrics.adoc:28]: Remove extra lines.

File: modules/psap-special-resource-operator-using-configmaps.adoc [hunk 7]
```diff
@@ -2,10 +2,11 @@
 //
 // * hardware_enablement/psap-special-resource-operator.adoc
 
+:_content-type: PROCEDURE
 [id="deploy-simple-kmod-using-configmap-chart"]
 = Building and running the simple-kmod SpecialResource by using a config map
 
-In this example, the simple-kmod kernel module is used to show how the SRO can manage a driver container which is defined in Helm chart templates stored in a config map.
+In this example, the simple-kmod kernel module is used to show how the Special Resource Operator (SRO) can manage a driver container which is defined in Helm chart templates stored in a config map.
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-using-configmaps.adoc:9]:
```suggestion
In this example, the simple-kmod kernel module shows how the Special Resource Operator (SRO) manages a driver container. The driver container is defined in the Helm chart templates that are stored in a config map.
```

This sentence is going to be hard to follow for someone who is not a native speaker. Suggestion above to break it up into two sentences for simpler reading and to apply minimalism.

File: modules/psap-special-resource-operator-using-configmaps.adoc [hunk 8]
```diff
@@ -270,7 +271,15 @@ spec:
 ----
 $ oc create -f simple-kmod-configmap.yaml
 ----
-+
+
+[NOTE]
+====
+If you want to remove the simple-kmod kernel module from the node, delete the simple-kmod `SpecialResource` API object using the `oc delete` command. The kernel module is unloaded when the driver container pod is deleted.
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-using-configmaps.adoc:277]:
```suggestion
To remove the simple-kmod kernel module from the node, delete the simple-kmod `SpecialResource` API object using the `oc delete` command. The kernel module is unloaded when the driver container pod is deleted.
```
Applying minimalism principles.

File: modules/psap-special-resource-operator-using-manifests.adoc [hunk 9]
```diff
@@ -2,10 +2,11 @@
 //
 // * hardware_enablement/psap-special-resource-operator.adoc
 
+:_content-type: PROCEDURE
 [id="deploy-simple-kmod-using-local-chart_{context}"]
 = Building and running the simple-kmod SpecialResource by using the templates from the SRO image
 
-The SRO image contains a local repository of Helm charts including the templates for deploying the simple-kmod kernel module. In this example, the simple-kmod kernel module is used to show how the SRO can manage a driver container that is defined in the internal SRO repository.
+The Special Resource Operator(SRO) image contains a local repository of Helm charts including the templates for deploying the simple-kmod kernel module. In this example, the simple-kmod kernel module is used to show how the SRO can manage a driver container that is defined in the internal SRO repository.
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-using-manifests.adoc:9]:
```suggestion
The Special Resource Operator(SRO) image contains a local repository of Helm charts, including the templates for deploying the simple-kmod kernel module. In this example, the simple-kmod kernel module shows how the SRO can manage a driver container that is defined in the internal SRO repository.
```

File: modules/psap-special-resource-operator-using-manifests.adoc [hunk 10]
```diff
@@ -53,10 +54,17 @@ spec:
 ----
 $ oc create -f simple-kmod-local.yaml
 ----
-+
+
+[NOTE]
+====
+If you want to remove the simple-kmod kernel module from the node, delete the simple-kmod `SpecialResource` API object using the `oc delete` command. The kernel module is unloaded when the driver container pod is deleted.
```
Comment (sagidlow) [inline modules/psap-special-resource-operator-using-manifests.adoc:60]:
```suggestion
To remove the simple-kmod kernel module from the node, delete the simple-kmod `SpecialResource` API object using the `oc delete` command. The kernel module is unloaded when the driver container pod is deleted.
```

**Review by sagidlow — COMMENTED** [review]
Let me know if you have any questions :)

File: modules/psap-special-resource-operator-metrics.adoc (hunk 5 above)
Comment (cherraz) [reply modules/psap-special-resource-operator-metrics.adoc:6]: Agree, but all the files in this assembly are called "psap-" and then the real id, so I've followed that convention to keep things aligned.

File: modules/psap-special-resource-operator-using-configmaps.adoc (hunk 7 above)
Comment (cherraz) [reply modules/psap-special-resource-operator-using-configmaps.adoc:9]: Agree. I've not tried to reword/rework these sentences, just made some tweaks I needed for my changes to work.  I'll apply this and the rest of the rewording suggestions, they will make the assembly better. 

Thanks!

File: modules/psap-special-resource-operator-metrics.adoc (hunk 4 above)
Comment (cherraz) [reply modules/psap-special-resource-operator-metrics.adoc:7]: Thank you!

**Comment by cherraz** [comment]
Hi @wabouhamad , @cdvultur  please someone review this PR and let me know if you find issues, or give me a lgtm otherwise.  Thank you!
